src/lib.rs

```diff
@@ -78,3 +82,65 @@ pub fn from_value<T: serde::de::DeserializeOwned>(value: JsValue) -> Result<T> {
 pub fn to_value<T: serde::ser::Serialize + ?Sized>(value: &T) -> Result<JsValue> {
     value.serialize(&Serializer::new())
 }
+
+/// A wrapper around a [`JsValue`] (or anything that can be cast into one)
+/// that makes it pass through serialization and deserialization unchanged.
+///
+/// # Example
+/// ```rust
+/// use serde_wasm_bindgen::{PreservedValue, to_value};
+/// #[derive(serde::Serialize)]
+/// struct MyStruct {
+///     int_field: i32,
+///     js_field: PreservedValue<JsValue>,
+/// }
+/// let big_array = js_sys::Int8Array::new_with_length(1000);
+/// let s = MyStruct {
+///     int_field: 5,
+///     js_field: PreservedValue(big_array.into()),
+/// };
+///
+/// // Will return a JsValue representing an object with two fields (`int_field` and `js_field`).
+/// // `js_field` will be an `Int8Array` pointing to the same underlying JavaScript object as `big_array`.
+/// to_value(&s);
+/// ```
+#[derive(Clone, Debug, PartialEq)]
+pub struct PreservedValue<T: JsCast>(pub T);
+
+// Some arbitrary string that no one will collide with unless they try.
+pub(crate) const PRESERVED_VALUE_MAGIC: &str = "1fc430ca-5b7f-4295-92de-33cf2b145d38";
+
+#[derive(serde::Serialize, serde::Deserialize)]
+#[serde(rename = "1fc430ca-5b7f-4295-92de-33cf2b145d38")]
+struct PreservedValueWrapper(u32);
+
+impl<'de, T: JsCast> serde::Deserialize<'de> for PreservedValue<T> {
```

Comment:
Hmm if you're still implementing `Serialize`/`Deserialize` by hand instead of using `serde(try_from)`/`serde(into)` or `serde(remote)`, then the wrapper only adds unnecessary indirection.

I meant to add it as a way to avoid complexity of writing those implementations by hand.

Reply:
Having the wrapper helps a little by avoiding the `Visitor` boilerplate. That savings could increase slightly if we also add a `TransferableValue` feature (which I will open up an issue to describe). But I'm happy to go either way, just let me know